python/ray/tests/test_concurrency_group.py

```diff
@@ -179,6 +180,28 @@ def ping(self):
     print(ray.get(n.ping.remote()))
 
 
+def test_invalid_concurrency_group():
+    """Verify that when a concurrency group has max concurrency set to 0,
+    an error is raised when the actor is created.
+    """
+
+    script = """
+import ray
+
+ray.init()
+
+@ray.remote(concurrency_groups={"io": 0, "compute": 0})
+class A:
+    def __init__(self):
+        pass
+
+actor = A.remote()
+    """
+
+    output = run_string_as_driver(script)
+    assert "max_concurrency must be greater than 0" in output
```

Comment:
The error message may not be clear enough, but it's difficult to provide a better one in `BoundedExecutor`. I believe most users won't set a group's max concurrency to 0, and if they do, the error message should be sufficient for debugging.

A possible follow up is to make it fail fast. For example, throw an error in Python.

Reply:
this should be fine for now